File: Java/백준/Silver/2960. 에라토스테네스의 체/에라토스테네스의 체.java
import java.util.*;
import java.io.*;

public class Main {

    static BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
    static StringTokenizer st;
    static boolean[] sieve;

    public static void main(String[] args) throws IOException {

        int n, k, cnt = 0, ans = 0;
        st = new StringTokenizer(br.readLine());
        n = Integer.parseInt(st.nextToken());
        k = Integer.parseInt(st.nextToken());

        sieve = new boolean[n + 1]; // 기본값 false
        sieve[0] = true;
        sieve[1] = true;

        for (int i = 2; i <= n; i++) {
            if (!sieve[i]) {
                cnt++;
                if (cnt == k)
                    ans = i;
                for (int j = i * 2; j <= n; j += i) {
                    if (!sieve[j]) {
                        sieve[j] = true; // 합성수 제거
                        cnt++;
                        if (cnt == k)
                            ans = j;
                    }
                }
            }
        }

        System.out.println(ans);
    }
}

File: PyPy3/백준/Silver/2960. 에라토스테네스의 체/에라토스테네스의 체.py
from sys import stdin

input = stdin.readline

n, k = map(int, input().split())

sieve = [False, False] + [True] * (n - 1)

cnt, ans = 0, 0
for i in range(2, n + 1):
    if sieve[i]:
        cnt += 1
        if cnt == k:
            ans = i
        for j in range(i * 2, n + 1, i):
            if sieve[j]:
                sieve[j] = False  # 합성수 제거
                cnt += 1
                if cnt == k:
                    ans = j

print(ans)
